airflow/models/baseoperator.py

```diff
@@ -1218,15 +1225,17 @@ def __deepcopy__(self, memo):
         result.__instantiated = self.__instantiated
         return result
 
-    def __getstate__(self):
+    def __getstate__(self) -> dict:
         state = dict(self.__dict__)
-        del state["_log"]
+        state["_log"] = self._log.name if self._log else None
```

Comment:
I don’t think we need to record `_log.name` at all? It is easily reconstructable from `_log_config_logger_name` and `_logger_name`.

Reply:
Here I want to store whenever a logger has already been initialized or not (`_log`is not initialized by default anymore with this PR). If not, I don't restore it in setstate.

Instead of a boolean, storing the name covers also the case where `_log` is set up in any other way than by the LoggingMixing. Right now, no operators is setting `_log` directly, but other classes in Airflow codebase do.

If we assume the LoggingMixin always define the `_log` for operators, then I can indeed store a flag instead of the name.

Reply:
> We can simply always set `self._log` to None and rely on logic in LoggingMixin to build that logger object when needed.

Updated the PR. It's indeed simpler like this. Thanks !